nixos/modules/system/activation/switch-to-configuration.pl

```diff
@@ -385,21 +392,21 @@ sub compare_units {
 
 # Called when a unit exists in both the old systemd and the new system and the units
 # differ. This takes care of what units are to be stopped, restarted, reloaded, started, and skipped.
-sub handle_modified_unit {
+sub handle_modified_unit { ## no critic(Subroutines::ProhibitManyArgs, Subroutines::ProhibitExcessComplexity)
     my ($unit, $base_name, $new_unit_file, $new_unit_info, $active_cur, $units_to_stop, $units_to_start, $units_to_reload, $units_to_restart, $units_to_skip) = @_;
 
-    if ($unit eq "sysinit.target" || $unit eq "basic.target" || $unit eq "multi-user.target" || $unit eq "graphical.target" || $unit =~ /\.path$/ || $unit =~ /\.slice$/) {
+    if ($unit eq "sysinit.target" || $unit eq "basic.target" || $unit eq "multi-user.target" || $unit eq "graphical.target" || $unit =~ /\.path$/msx || $unit =~ /\.slice$/msx) {
         # Do nothing.  These cannot be restarted directly.
 
         # Slices and Paths don't have to be restarted since
         # properties (resource limits and inotify watches)
         # seem to get applied on daemon-reload.
-    } elsif ($unit =~ /\.mount$/) {
+    } elsif ($unit =~ /\.mount$/msx) {
         # Reload the changed mount unit to force a remount.
         # FIXME: only reload when Options= changed, restart otherwise
         $units_to_reload->{$unit} = 1;
         record_unit($reload_list_file, $unit);
-    } elsif ($unit =~ /\.socket$/) {
+    } elsif ($unit =~ /\.socket$/msx) {
```

Comment:
Not a fan of the `/msx` everywhere, especially on clearly single-line strings. Why is that required?

Reply:
OK, it's fine